File: python/Graph.py
class Graph:

	def __init__(self, directed = True):
		self.directed = directed
		self.nodes = []
		self.edges = []


	def addNode(self, id):
		self.nodes.append(id)


	def addEdge(self, source, target, weight=1):
		self.edges.append({
			"source": source,
			"target": target,
			"weight": weight
		})

	def totalNodes(self):
		return len(self.nodes)


	def getNeighbours(self, id):
		neighbours = []
		for edge in self.edges:
			if edge["source"] == id:
				neighbours.append({
					"id": edge["target"],
					"weight": edge["weight"]
				})
			
			if(not self.directed and edge["target"] == id):
				neighbours.append({
					"id": edge["source"],
					"weight": edge["weight"]
				})


		return neighbours
	

File: js/Graph.js
export class Graph {

	constructor(directed = true) {
		this.directed = directed
		this.nodes = []
		this.edges = []
	}

	addNode(id) {
		this.nodes.push(id)
	}

	addEdge(source, target, weight=1) {
		this.edges.push({source, target, weight})
	}
	getNeighbours(id) {
		const neighbours = []
		for(let edge of this.edges) {
			if(edge.source == id) {
				neighbours.push({
					id: edge.target,
					weight: edge.weight
				})
			}
			if(!this.directed && edge.target == id) {
				neighbours.push({
					id: edge.source,
					weight: edge.weight
				})
			}
		}
		return neighbours
	}
	totalNodes() {return this.nodes.length}
	getEdges() {return this.edges}
	getNodes() {return this.nodes}
}


// removeNode(id) {
// 	const index = this.nodes.indexOf(id)
// 	this.nodes.splice(index, 1)
// }
// removeEdge(source, target) {
// 	const index = this._getIndexOfConnection(source, target)
// 	this.edges.splice(index, 1)
// }

// _getIndexOfConnection(source, target) {
// 	return this.nodes.findIndex(edge => edge.source == source && edge.target == target)
// }